src/logics/Logic.cc

```diff
@@ -944,6 +944,27 @@ PTRef Logic::mkEq(vec<PTRef>& args) {
     return mkFun(eq_sym, args);
 }
 
+PTRef Logic::mkDistinct(vec<PTRef>& args) {
+    if (args.size() == 0) return getTerm_false(); // Is this correct?
+    if (args.size() == 1) return getTerm_true(); // Is this correct?
+    sort(args);
+    int i = 1;
+    int j = 0;
+    for (; i < args.size(); i++, j++) {
+        if (args[j] == args[i]) {
+            return getTerm_false();
+        }
+    }
+    SymRef diseq_sym = term_store.lookupSymbol(tk_distinct, args);
+    // The boolean distinctness is either xor or false
+    if (hasSortBool(args[0])) {
```

Comment:
In mkEq this check checks the sort of the symbol, not the first argument.
Maybe it should be like that also here?

Reply:
Interestingly this is a bug in mkEq (shielded by the conditions inside).  I'm going to fix it the same way it is now for distinct.

Reply:
True, I have just discovered this as well. It does the unnecessary checks also for real equality.